app/src/main/java/com/example/recipeapp/fragments/RecipeSearchFragment.java

```diff
@@ -33,16 +64,69 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
     @Override
     public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
         super.onViewCreated(view, savedInstanceState);
+        recipes = new ArrayList<>();
+        adapter = new RecipeSearchAdapter(getContext(),recipes);
+        binding.rvRecipes.setAdapter(adapter);
+        GridLayoutManager gridLayoutManager = new GridLayoutManager(getContext(), 2);
+        binding.rvRecipes.setLayoutManager(gridLayoutManager);
+        // TODO: Implement Endless Scrolling and Refresh
         binding.svSearch.setOnQueryTextListener(new SearchView.OnQueryTextListener() {
             @Override
             public boolean onQueryTextSubmit(String query) {
                 Log.i(TAG, query);
+                try {
+                    getRecipes(query);
+                } catch (IOException e) {
+                    e.printStackTrace();
+                }
                 return true;
             }
 
             @Override
             public boolean onQueryTextChange(String newText) {
-                return false;
+                Log.i(TAG, newText);
+                try {
+                    getRecipes(newText);
+                } catch (IOException e) {
+                    e.printStackTrace();
+                }
+                return true;
+
+            }
+        });
+    }
+
+    private void getRecipes(String query) throws IOException {
+        AsyncHttpClient client = new AsyncHttpClient();
+        RequestHeaders headers = new RequestHeaders();
+        headers.put("X-RapidAPI-Key", getString(R.string.Nutrition_API_Key));
+        headers.put("X-RapidAPI-Host","spoonacular-recipe-food-nutrition-v1.p.rapidapi.com");
+
+        RequestParams params = new RequestParams();
+        params.put("query",query);
+        params.put("addRecipeInformation","true");
+        params.put("instructionsRequired","true");
+        params.put("sortDirection","asc");
+
+
+        client.get("https://spoonacular-recipe-food-nutrition-v1.p.rapidapi.com/recipes/complexSearch",headers,params, new JsonHttpResponseHandler() {
```

Comment:
I'd use constants to build this URL.

Reply:
I made a RecipeClient class in a future commit that contains urls as constants.